Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Dashboard/Route.hs

```
@@ -0,0 +1,62 @@
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.Dashboard.Route where
+
+import qualified Data.List.NonEmpty as NE
+import qualified Domain.Types.Booking.Type as DRB
+import qualified Domain.Types.Merchant as DM
+import qualified Domain.Types.Ride as DRide
+import Environment
+import Kernel.External.Maps (LatLong (..))
+import qualified Kernel.External.Maps as Maps
+import Kernel.External.Maps.Interface.Types
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto (runInReplica)
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import qualified Storage.CachedQueries.Merchant as QMerchant
+import qualified Storage.Queries.Booking as QRB
+import qualified Storage.Queries.Ride as QRide
+import Tools.Maps (getTripRoutes)
+
+newtype TripRouteReq = TripRouteReq
+  { pickupLocation :: Maps.LatLong
+  }
+  deriving (Show, ToJSON, FromJSON, Generic, ToSchema)
+
+mkGetLocation :: ShortId DM.Merchant -> Id DRide.Ride -> TripRouteReq -> Flow GetRoutesResp
+mkGetLocation shortMerchantId rideId req = do
+  merchant <-
+    QMerchant.findByShortId shortMerchantId
+      >>= fromMaybeM (MerchantDoesNotExist shortMerchantId.getShortId)
+  ride <- runInReplica $ QRide.findById (cast rideId) >>= fromMaybeM (RideDoesNotExist rideId.getId)
+  booking <- runInReplica $ QRB.findById ride.bookingId >>= fromMaybeM (BookingDoesNotExist ride.bookingId.getId)
+  let mbToLocation = case booking.bookingDetails of
+        DRB.OneWayDetails details -> Just details.toLocation
+        DRB.RentalDetails _ -> Nothing
+        _ -> Nothing
+  let bookingLocation = fromJust mbToLocation
```

please don't use partial function in production code
its better to throw an appropriate error:
`bookingLocation <- mbToLocation & fromMaybeM "Drop location does not exist for this ride"`